Tests/MountebankSwiftTests/Models/RecreatableTests.swift

```diff
@@ -0,0 +1,315 @@
+import InlineSnapshotTesting
+import XCTest
+@testable import MountebankSwift
+
+final class RecreatableTests: XCTestCase {
+    struct MyStruct: Recreatable {
+        let foo: String?
+        let bar: [Int]
+        let enumValue: MyEnum
+        let json: JSON?
+
+        var recreatable: String {
+            structSwiftString([
+                ("foo", foo),
+                ("bar", bar),
+                ("enumValue", enumValue),
+                ("json", json),
+            ])
+        }
+    }
+
+    indirect enum MyEnum: Recreatable {
+        case simple
+        case associatedValue(String)
+        case namedAssociatedValues(foo: String, bar: String)
+        case structValue(structValue: MyStruct)
+
+        var recreatable: String {
+            switch self {
+            case .simple:
+                return enumSwiftString()
+            case .associatedValue(let string):
+                return enumSwiftString([string])
+            case .namedAssociatedValues(let foo, let bar):
+                return enumSwiftString([("foo", foo), ("bar", bar)])
+            case .structValue(let structValue):
+                return enumSwiftString([("structValue", structValue)])
+            }
+        }
+    }
+
+    func testString() {
+        assertInlineSnapshot(of: "some string value".recreatable, as: .lines) {
+            """
+            "some string value"
+            """
+        }
+    }
+
+    func testInt() {
+        assertInlineSnapshot(of: 42.recreatable, as: .lines) {
+            """
+            42
+            """
+        }
+    }
+
+    func testDouble() {
+        assertInlineSnapshot(of: 42.0.recreatable, as: .lines) {
+            """
+            42.0
+            """
+        }
+    }
+
+    func testBool() {
+        assertInlineSnapshot(of: true.recreatable, as: .lines) {
+            """
+            true
+            """
+        }
+    }
+
+    func testData() {
+        let stringValue = "some-string"
+        // swiftlint:disable:next force_unwrapping
+        let data: Data = stringValue.data(using: .utf8)!
+        assertInlineSnapshot(of: data.recreatable, as: .lines) {
+            """
+            Data(base64Encoded: "c29tZS1zdHJpbmc=")!
+            """
+        }
+        XCTAssertEqual(
+            // swiftlint:disable:next force_unwrapping
+            String(bytes: Data(base64Encoded: "c29tZS1zdHJpbmc=")!, encoding: .utf8),
+            stringValue
+        )
+    }
+
+    func testEnum() {
+        assertInlineSnapshot(of: MyEnum.simple.recreatable, as: .lines) {
+            """
+            .simple
```

Comment:
should this not be `MyEnum.simple`? Same for other enums.

Reply:
I considered this, but the compiler can always infer the type and this is a lot cleaner.